utils/trivia.py

```diff
@@ -78,31 +78,62 @@ class Trivia:
+        self.qAsked = QuestionAnswer("Question asked", "Answer asked")
+
+        # This line prepopulates the question list and is only here to facilitate
+        # ease of development. Had development continued far enough, this line would
+        # be replaced with code that would read in the questions and answers from an
+        # external source.
         self.qlist = QuestionList([QuestionAnswer("Question 1", "Answer 1")])
+        
         self.inProgress = False
+        
+    ## This function is responsible for starting the trivia game.
+    # Return non-zero value to indicate failure
+    # Return 0 to indicate success
+    def start_game(self) -> int:
+        if __debug__:
+            assert type(self.inProgress) is bool, "The inProgress variable must be of type boolean"
+            assert self.inProgress == False, "inProgress must be False when start_game is called"
+
+            # Remainder of function not yet implemented, this is just a stub.
 
-    # This function is responsible for starting the trivia game.
-    def start(self) -> int:
-        # Return non-zero value to indicate failure
-        # Return 0 to indicate success
         return 0
 
-    # This function is responsible for ending the trivia game.
-    def end(self) -> int:
-        # Return non-zero value to indicate failure
-        # Return 0 to indicate success
+    ## This function is responsible for ending the trivia game.
+    # Return non-zero value to indicate failure
+    # Return 0 to indicate success
+    def end_game(self) -> int:
+        if __debug__:
+            assert type(self.inProgress) is bool, "The inProgress variable must be of type boolean"
+            assert self.inProgress == True, "inProgress must be True when the Trivia object is initialized"
+
+            # Remainder of function not yet implemented, this is just a stub.
+
         return 0
 
-    def check(self, answer) -> int:
-        # Return non-zero value to indicate failure
-        # Return 0 to indicate success
+    ## This function is responsible for checking submitted answers
+    # to see if they are correct or not.
+    # Return non-zero value to indicate failure
+    # Return 0 to indicate success
+    def check_answer(self, answer) -> int:
+        if __debug__:
+            assert type(answer) is str, "The answer argument must be of type str"
+
         if (self.qAsked.answer == answer):
             return 0
         else:
             return -1
         
 
-    # This function is responsible for returning the status of the
+    ## This function is responsible for returning the status of the
     # trivia game.
-    def status(self) -> str:
-        return "Trivia Status!"
+    def get_game_status(self) -> str:
+        if __debug__:
+            assert type(self.inProgress) is bool, "inProgress must be of type bool"
+
+        if (self.inProgress):
+            return "Trivia game in progress"
+        else:
+            return "Trivia game is not in progress"
```

Comment:
Note to self for test updates that the return value for this method has changed.